scripts/screens/EventEditScreen.py

```diff
@@ -695,10 +735,42 @@ def handle_event(self, event):
 
         # SEARCHING
         if event.type == pygame.KEYDOWN and event.key == pygame.K_RETURN:
-            self.search_text = self.event_search.get_text()
-            self.create_event_display(
-                event_type=self.chosen_type, biome=self.chosen_biome
-            )
+            if self.event_search.is_focused:
+                self.search_text = self.event_search.get_text()
+                self.create_event_display(
+                    event_type=self.chosen_type, biome=self.chosen_biome
+                )
+            # FUTURE EVENT IDS
+            elif self.future_element["include_entry"].is_focused:
+                new_id = self.future_element["include_entry"].get_text()
+                block_info = self.get_selected_block_info()["pool"]["event_id"]
+                if new_id not in block_info:
+                    block_info.append(new_id)
+                else:
+                    block_info.remove(new_id)
+                text = ""
+                for id in block_info:
+                    text += f"'{id}'<br>"
+                self.future_element["include_display"].set_text(text)
+                self.future_element["include_entry"].set_text("")
+                self.editor_container.on_contained_elements_changed(
+                    self.future_element["include_display"]
+                )
+            elif self.future_element["exclude_entry"].is_focused:
+                new_id = self.future_element["exclude_entry"].get_text()
+                block_info = self.get_selected_block_info()["pool"]["excluded_event_id"]
+                if new_id not in block_info:
+                    block_info.append(new_id)
+                else:
+                    block_info.remove(new_id)
+                text = ""
+                for id in block_info:
+                    text += f"'{id}'<br>"
+                self.future_element["exclude_display"].set_text(text)
+                self.future_element["exclude_entry"].set_text("")
+                self.editor_container.on_contained_elements_changed(
+                    self.future_element["exclude_display"]
+                )
```

Comment:
Could all this be moved into a function? Just because handle_event is already horrifyingly enormous in this screen.